### DISCUSSION TITLE
Wiring

### DISCUSSION THREAD

**Discussion by sspieser** [doudar/SmartSpin2k#109] [Q&A]
Hello,
Do you have a wiring diagram ?

I'm wondering : what is the output of the LM2596, 5v or 12V? I guess 5V... but then, what is the TMC2225 :VMOT +12V ? An output or an input ?

Thanks for your help,
S

**Comment by doudar** [comment]
For the non-PCB, these should work:
https://github.com/doudar/SmartSpin2k/blob/master/Pictures/SmartSpin2k_Esp32_Connections.jpg
https://github.com/doudar/SmartSpin2k/blob/master/Pictures/SmartSpin2k_TMC_Connections.jpg

Additionally you may find my assembly pictures helpful although I haven't had time to add documentation to them yet:
https://github.com/doudar/SmartSpin2k/tree/master/Pictures/Assembly

**Comment by sspieser** [comment]
Thanks; I need two clarifications, one : VMOT 12V ln the TMC2225, where
does that go? If this is input, well, I have a 24V power... If output,
well, I do not see where to go...

Second point, the UART wiring: I need to adf a 1k before plugging both
ESP32 pins at the same TMC pin? Which one then?

Thanks a lot!



Le sam. 6 févr. 2021 à 19:01, Anthony Doud <notifications@github.com> a
écrit :

**Comment by doudar** [comment]
The 1k ohm resistor goes between pin 12 and the uart pin on the tmc. Pin 14 no resistor, connect directly to the uart pin.

You are correct, VMOT is input power and can take up to +24v

**Comment by sspieser** [comment]
So, VMOT : you plug it with the output of your SMartspin2K power supply ?

For the UART : whih one one the TMA2225 is the UART pin you wire : if I read the datasheet well, there's 2 UART pin, depending on the mode you use it... probable you wire the upper one ?

Thx again, Seb

**Comment by doudar** [comment]
> So, VMOT : you plug it with the output of your SMartspin2K power supply ?
> 
> For the UART : whih one one the TMA2225 is the UART pin you wire : if I read the datasheet well, there's 2 UART pin, depending on the mode you use it... probable you wire the upper one ?
> 
> Thx again, Seb

VMOT is the power you're switching - yes, the voltage coming from your wall power supply. The other comes from your 5v power supply.  

Here's the PCB schematic if it helps:
![image](https://user-images.githubusercontent.com/17362216/107148741-7d77c700-691a-11eb-9343-c9de94c3ed52.png)

The UART pin can differ depending on the configuration of the stepstick. The ones from BigTreeTech are configured as above.

**Comment by sspieser** [comment]
Thanks a lot !
So I just began the wiring : I'm trying to power up the ESP32.... Plugged +5V, GND... but I cannot get the Web Server...

**Comment by doudar** [comment]
Hard to say exactly. 

Does it work if you're plugged in via USB only?

Does the LED blink indicating it's on? A new version of SmartSpin2K was just released so it may be doing an OTA update (light flashing rapidly) which could take a couple minutes after powerup.

**Comment by sspieser** [comment]
It was on with the red light, but not blinking at all... Sure I'l do it
again with USB, and with the PC... thanks, I'll keep you posted :-)

Le dim. 7 févr. 2021 à 22:07, Anthony Doud <notifications@github.com> a
écrit :

**Comment by doudar** [comment]
Good luck! 

Yes, the blue light should start blinking while it's waiting for a client (Bluetooth) connection.

**Comment by doudar** [comment]
Any Updates @sspieser ?

**Comment by sspieser** [comment]
Hello, yes : it works, I can get to the ESP32 WEB server !
I have found that my ESP32 devkit (now I use this one: [https://www.az-delivery.de/fr/products/esp-32-dev-kit-c-v4](https://www.az-delivery.de/fr/products/esp-32-dev-kit-c-v4)) had a third GND pin... using this one works!

I noticed thus a weird thing : while using the WEB application from my phone, I often lose any WIFI signal on the phone and lose Internet connection... just as if the ESP32 was scrambling things...

But anyway, it works, I paired my pedals ; now I have to keep on wiring but I am not at home for some days; I'll keep you posted and put pictures as soon as I can :-)

**Comment by sspieser** [comment]
Hello again, looking at the chart above, I'm wondering : TMC2225(1) pin 5 is linked to the ESP32 D27 pin ; but this link is missing from the diagram on the hme page : 
![image](https://user-images.githubusercontent.com/14963451/108117521-4b463200-709d-11eb-9dfd-4f8261f70b0d.png)

Do you confirm pin 5 and D27 must be linked also ?

**Comment by doudar** [comment]
The PCB has extra connections on it in case I ever switch to a different stepper driver. 

I believe ms1, ms2 and SLP can all be omitted/unconnected.

**Comment by sspieser** [comment]
Thanks.... not sure to understand your last answer ;-)
But I will keep on wiring stuff: +24V for the stepper, and the UART/ 1k stuff ; so far, waiting for wires...

**Comment by sspieser** [comment]
I think I'm almost done, just the UART wiring missing ! But then, testing... so far, unable to get in touch with the WEB application...

![DSC_0578](https://user-images.githubusercontent.com/14963451/108524965-e4f52580-72cf-11eb-8ca8-01472b0d9167.JPG)

**Comment by doudar** [comment]
Starting to look good @sspieser ! 

UART is not required for it to work in a basic mode, so you should be able to test it just fine. 

I'm not certain what the issue may be with the webserver. It basically "just works" for me. The only thing I can think of is maybe a bad antenna on your board? 

Do you see the SmartSpin2K access point? If not it means it is successfully connected to your router. Try to do manual firmware update https://github.com/doudar/OTAUpdates/blob/main/firmware.bin on the unit and connect it to your local network, it may help me troubleshoot your issue. 

The easiest is to upload it via USB cable with (the brand new) https://github.com/SmartSpin2K/SmartSpin2kFlasher/releases/tag/v0.0.1

**Comment by jkbaker4** [comment]
I am wiring my setup this evening.  After looking at the diagrams many times I am assuming that the 24v coming out of the power supply will power the TMC2225 as well as the LM2596.  If that is true, then the TMC2225 will be adjusted to provide 12V to the stepper motor, then the LM2596 will be adjusted to provide 5v to the ESP32 and TMC2225.  Is that correct?

**Reply by fireyeti** [reply]
@jkbaker4 yes it sounds like you’ve got it correct. I believe you do not have to adjust the TMC2225 to provide 12V to the stepper, as the wiring diagram was written prior to the power supply increase to 24V from the 12V supply which was used in early builds.

**Comment by posssson** [comment]
Hello, thank you for this projet I made all links but on the PCB figure you connect PIN D26 and D27 but not in the ESP32 picture. Why?

**Comment by doudar** [comment]
> Hello, thank you for this projet I made all links but on the PCB figure you connect PIN D26 and D27 but not in the ESP32 picture. Why? 

Good question!

The PCB has a few extra wire traces just for potential future capabilities with different stepper drivers. 

They are not currently used and have no plans for use.

**Reply by doudar** [reply]
Yes, both the ESP32 devkit and the TMC step sticks have an internal 3.3v supply voltage regulator.

**Comment by chinasaur** [comment]
I have a few wiring questions as well. It looks like the D26/D27 question is already answered. :)

I have a slightly different TMC2225 board, which has separate TX/RX pins marked for UART. Maybe they're just the same pin internally? I don't understand the 1kOhm resistor; it's for voltage protection? Have you tried running TMC2225 at 3.3V?

**Reply by doudar** [reply]
The resistor is actually for voltage differentiation between rx and tx signals so they are a different level and can share the same line. Some step sticks have the resistor on the stick, most do not.

**Reply by chinasaur** [reply]
Got it, thanks. And I see some references online to the UART pins being 3.3V regardless of whether Vio is 3.3V or 5V, is that right?

**Reply by doudar** [reply]
See above...My first reply went to the wrong thread ^^
